cibuildwheel/options.py

```diff
@@ -137,20 +140,121 @@ class Override:
+    pass
 
 
-class ConfigOptionError(KeyError):
-    pass
+class OptionFormat:
+    """
+    Base class for option format specifiers. These objects describe how values
+    can be parsed from rich TOML values and how they're merged together.
+    """
+
+    class NotSupported(Exception):
+        pass
+
+    def format_list(self, value: SettingList) -> str:  # noqa: ARG002
+        raise OptionFormat.NotSupported
+
+    def format_table(self, table: SettingTable) -> str:  # noqa: ARG002
+        raise OptionFormat.NotSupported
+
+    def merge_values(self, before: str, after: str) -> str:  # noqa: ARG002
+        raise OptionFormat.NotSupported
+
+
+class ListFormat(OptionFormat):
+    """
+    A format that joins lists with a separator.
+    """
+
+    def __init__(self, sep: str) -> None:
+        self.sep = sep
+
+    def format_list(self, value: SettingList) -> str:
+        return self.sep.join(str(v) for v in value)
+
+    def merge_values(self, before: str, after: str) -> str:
+        return f"{before}{self.sep}{after}"
+
+
+class ShlexTableFormat(OptionFormat):
+    """
+    The standard table format uses shlex.quote to quote values and shlex.split
+    to unquote and split them. When merging values, keys in before are
+    replaced by keys in after.
+    """
+
+    def __init__(self, sep: str = " ", pair_sep: str = "=", allow_merge: bool = True) -> None:
+        self.sep = sep
+        self.pair_sep = pair_sep
+        self.allow_merge = allow_merge
+
+    def format_table(self, table: SettingTable) -> str:
+        assignments: list[tuple[str, str]] = []
+
+        for k, v in table.items():
+            if shlex.split(k) != [k]:
+                msg = f"Invalid table key: {k}"
+                raise OptionsReaderError(msg)
+
+            if isinstance(v, str):
+                assignments.append((k, v))
+            elif isinstance(v, Sequence):
+                for inner_v in v:
+                    assignments.append((k, str(inner_v)))
+            else:
+                assignments.append((k, str(v)))
+
+        return self.sep.join(f"{k}{self.pair_sep}{shlex.quote(v)}" for k, v in assignments)
+
+    def merge_values(self, before: str, after: str) -> str:
+        if not self.allow_merge:
+            raise OptionFormat.NotSupported
+
+        before_dict = self.parse_table(before)
+        after_dict = self.parse_table(after)
+
+        return self.format_table({**before_dict, **after_dict})
+
+    def parse_table(self, value: str) -> Mapping[str, str | Sequence[str]]:
+        assignments: list[tuple[str, str]] = []
+
+        for assignment_str in shlex.split(value):
+            key, _, value = assignment_str.partition(self.pair_sep)
+
+            if not value:
+                msg = f"malformed option with value {assignment_str!r}"
+                raise OptionsReaderError(msg)
+
+            assignments.append((key, value))
+
+        result: dict[str, str | list[str]] = {}
+
+        for key, value in assignments:
```

Comment:
```suggestion
        for key, val in assignments:
```

`value` is defined above.

Reply:
whoops. yeah, good point